app/models/katello/glue/candlepin/pool.rb

```diff
@@ -93,24 +123,13 @@ def provider?(organization)
 
       # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
       # rubocop:disable Metrics/CyclomaticComplexity
-      # rubocop:disable Metrics/PerceivedComplexity
       def import_data(index_hosts_and_activation_keys = false)
         pool_attributes = {}.with_indifferent_access
         pool_json = self.backend_data
 
-        self.organization ||= Organization.find_by(:label => pool_json['owner']['key'])
         product_attributes = pool_json["productAttributes"] + pool_json["attributes"]
-
         product_attributes.map { |attr| pool_attributes[attr["name"].underscore.to_sym] = attr["value"] }
 
-        if pool_json["sourceStackId"]
-          subscription = Pool.stacking_subscription(pool_json['owner']['key'], pool_json["sourceStackId"])
-        else
-          subscription = ::Katello::Subscription.find_by(:cp_id => pool_json["productId"])
```

Comment:
This is the culprit responsible for assigning the given pool to a subscription in a different org. the productId is not unique to an org so the first one it finds is the lucky winner.

Reply:
i remember soooo many bugs in the past with this same root cause!

Reply:
iirc pre-candlepin 2.0 products were shared across owners, but as part of 2.0 they no longer were, so this is probably from around before then

Reply:
Makes perfect sense!